gensim/models/word2vec.py

```diff
@@ -1521,6 +1521,54 @@ def __iter__(self):
                         i += self.max_sentence_length
 
 
+class LineSentencePath(object):
```

Comment:
Personally I'd consider the name `PathLineSentences` more descriptive, but other may have an even better name.